File: src/main/java/de/esymetric/jerusalem/osmDataRepresentation/osm2ownMaps/longOsm2OwnMaps/LongOsmNodeID2OwnIDMap.java
package de.esymetric.jerusalem.osmDataRepresentation.osm2ownMaps.longOsm2OwnMaps;

public interface LongOsmNodeID2OwnIDMap {
    int getNumberOfUsedArrays();

    boolean put(long osmNodeID, int ownNodeID);

    int get(long osmNodeID);

    void close();

    void delete();

}

File: src/main/java/de/esymetric/jerusalem/osmDataRepresentation/osm2ownMaps/longOsm2OwnMaps/LongOsmNodeID2OwnIDMap.kt
package de.esymetric.jerusalem.osmDataRepresentation.osm2ownMaps.longOsm2OwnMaps

interface LongOsmNodeID2OwnIDMap {
    val numberOfUsedArrays: Int
    fun put(osmNodeID: Long, ownNodeID: Int): Boolean
    operator fun get(osmNodeID: Long): Int
    fun close()
    fun delete()
}